fdbserver/RocksDBCheckpointUtils.actor.cpp

```diff
@@ -539,8 +542,7 @@ ACTOR Future<Void> fetchCheckpointFile(Database cx,
 			wait(IAsyncFileSystem::filesystem()->deleteFile(localFile, true));
 			const int64_t flags = IAsyncFile::OPEN_ATOMIC_WRITE_AND_CREATE | IAsyncFile::OPEN_READWRITE |
 			                      IAsyncFile::OPEN_CREATE | IAsyncFile::OPEN_UNCACHED | IAsyncFile::OPEN_NO_AIO;
-			state int64_t offset = 0;
-			state Reference<IAsyncFile> asyncFile = wait(IAsyncFileSystem::filesystem()->open(localFile, flags, 0666));
+			wait(store(asyncFile, IAsyncFileSystem::filesystem()->open(localFile, flags, 0666)));
```

Comment:
I wonder why this kind of state variable code style is needed for `OPEN_FOR_IDE=ON` to work? In the past, the `OPEN_FOR_IDE=ON` also works fine in CLion without this change.

Reply:
The problem here is that `asyncFile` is being used in another scope (the catch block), which technically compiles in actors because the state variables are actually members of the actor class. If it weren't a state variable, though, then trying to declare it in the try block would make it inaccessible in the catch block.